GitUI/UserControls/FilterToolBar.cs

```diff
@@ -362,6 +356,21 @@ private void revisionGridFilter_FilterChanged(object? sender, FilterChangedEvent
                 }
             }
 
+            // Add to dropdown and settings, unless already included
+            string filter = tstxtRevisionFilter.Text;
+            if (!string.IsNullOrWhiteSpace(filter) && (tstxtRevisionFilter.Items.Count == 0 || filter != (string)tstxtRevisionFilter.Items[0]))
+            {
+                if (tstxtRevisionFilter.Items.Contains(filter))
+                {
+                    tstxtRevisionFilter.Items.Remove(filter);
+                }
+
+                tstxtRevisionFilter.Items.Insert(0, filter);
+                tstxtRevisionFilter.Text = filter;
+                AppSettings.RevisionFilterDropdowns = tstxtRevisionFilter.Items.Cast<object>()
+                    .Select(item => item.ToString()).Take(30).ToArray();
```

Comment:
Please make this at least a local const.

Reply:
ok